### ISSUE TITLE
Add ingress e2es as a merge blocker

### ISSUE THREAD

**Issue by bprashanth** [kubernetes/kubernetes#23416]
These tests run as [Feature] because they needed a seperate project with higher quota.
They needed a seperate project because it was difficult to isolate L7 leaks from L4 leaks and trace it down to a single root cause. 
We now have the ability to run in a different project and still be a part of something like gce-slow (right?), so we should do that.

**Comment by roberthbailey** [comment]
/cc @lavalamp 

Did ingress ever get enabled in any of the merge queue blocking test suites?

**Comment by bprashanth** [comment]
>We now have the ability to run in a different project and still be a part of something like gce-slow (right?), so we should do that.

It takes ~15m, and it uses same resources as type=lb (they won't collide, but figuring out which test leakes a firewall named "1o09u90" a 100 runs later and getting its logs is near impossible). Also they have higher quota requirements. So as long as I can run them in a different project like they currently run, adding them to gce-slow should be easy. 

It's currently running in ingress-gke/gce{version}, the plan was to move it somewhere before GA (not happening till 1.4).

**Comment by lavalamp** [comment]
@bprashanth New rule: presubmit test of some sort required before we can turn on a postsubmit test.

Otherwise we spend too much time when little things completely break the post submit tests.

It will be difficult to convince @fejta or me to make an exception, but you can try :)

**Comment by goltermann** [comment]
Is this required for 1.3?  It doesn't have a priority applied.

**Comment by bprashanth** [comment]
Ingress hasn't moved anywhere since 1.2, it's on hold. It will probably go GA in next-release at which time I'll do it. for now I'm just extending the existing e2es (https://github.com/kubernetes/kubernetes/pull/26821), removing milestone.

**Comment by fejta** [comment]
Presubmit ingress test sounds fabu

**Comment by bprashanth** [comment]
The problem with that is it takes 15m for the GCE l7 to start responding to health checks. I'm planning to test the nginx controller instead of the gce controller in presumbit, because that starts up in seconds and gives us a good signal about any api level changes; but it needs some polishing.

**Comment by fejta** [comment]
How about a fake L7LB for presubmit?

**Comment by bprashanth** [comment]
That's what I meant by nginx controller (closes you'll get to something that is usable in the real world, vs something more like an integration test with a fake gce implementation watching apiserver).

**Comment by fejta** [comment]
I like it!

**Comment by goltermann** [comment]
@bprashanth are these tests running continuously (whether or not they're in a merge blocking suite)?

**Comment by bprashanth** [comment]
there are GKE/GCE suites for head, 1.3 and 1.4

**Comment by bprashanth** [comment]
I meant there are suites that run back to back for each of those builds on jenkins clusters (http://kubekins.dls.corp.google.com/job/kubernetes-e2e-gce-ingress/)

**Comment by goltermann** [comment]
Closing as we have continuous testing for this area now.
